source/postprocess/matrix_statistics.cc

```diff
@@ -45,12 +46,13 @@ namespace
            << " MB." << std::endl;
 
     // output number of nonzero elements in matrix. Do so with 1000s separator
-    // since they are frequently large; this is done by using the empty
-    // string locale, but creating std::locale with an empty string causes problems
-    // on some platforms, so catch the exception and ignore
+    // since they are frequently large; this was previously done by using the empty
+    // string locale, but creating std::locale with an empty string caused problems
+    // on some platforms, so the functionaltity yo catch the exception and ignore
+    // is kept here, even though explicitly setting a facet should always work.
     try
       {
-        output.imbue(std::locale(""));
+        output.imbue(std::locale(std::locale(), new aspect::Utilities::ThousandSep));
```

Comment:
Ok, so this is a bit of a headache. If we create a local variable or static local variable `thousand_sep`, and pass its address, we end up with `malloc: *** error for object 0x100e629d0: pointer being freed was not allocated`. Basically, when we exit `get_stats()`, we end up triggering the destructor of `&thousand_sep` twice. Instead, according to http://stdcxx.apache.org/doc/stdlibug/25-2.html and http://www.cplusplus.com/reference/locale/ctype/ctype/ , I believe the correct thing is indeed to use `new`: the `std::locale(std::locale(), new aspect::Utilities::ThousandSep)` takes responsibility for the facet, handles the destruction of the facet itself, and thus we don't have a memory leak.

A similar thing happens in the `core.cc` code too, when `imbue(s)` is called: it tries to destroy the `std::locale(std::locale(), &thousand_sep)` but ends up trying to destroy the pointer twice because the `locale` is already responsible.